modules/auction-manager/src/lib.rs

```diff
@@ -359,35 +359,34 @@ impl<T: Config> Pallet<T> {
 			max_iterations
 		);
 
-		// start iterations to cancel collateral auctions
-		let mut iterator = <CollateralAuctions<T> as IterableStorageMapExtended<_, _>>::iter(max_iterations, start_key);
-
-		#[allow(clippy::while_let_on_iterator)]
-		while let Some((collateral_auction_id, _)) = iterator.next() {
-			if let (Some(collateral_auction), Some((_, last_bid_price))) = (
-				Self::collateral_auctions(collateral_auction_id),
-				Self::get_last_bid(collateral_auction_id),
-			) {
-				// if collateral auction has already been in reverse stage,
-				// should skip it.
-				if collateral_auction.in_reverse_stage(last_bid_price) {
-					continue;
+		if let Some(iterator_start_key) = start_key {
+			// start iterations to cancel collateral auctions
+			let iterator = <CollateralAuctions<T>>::iter_from(iterator_start_key);
+
+			for (collateral_auction_id, _) in iterator {
+				if let (Some(collateral_auction), Some((_, last_bid_price))) = (
+					Self::collateral_auctions(collateral_auction_id),
+					Self::get_last_bid(collateral_auction_id),
+				) {
+					// if collateral auction has already been in reverse stage,
+					// should skip it.
+					if collateral_auction.in_reverse_stage(last_bid_price) {
+						continue;
+					}
 				}
+				Self::submit_cancel_auction_tx(collateral_auction_id);
+				guard.extend_lock().map_err(|_| OffchainErr::OffchainLock)?;
 			}
-			Self::submit_cancel_auction_tx(collateral_auction_id);
-			guard.extend_lock().map_err(|_| OffchainErr::OffchainLock)?;
-		}
 
-		if iterator.finished {
 			to_be_continue.clear();
-		} else {
-			to_be_continue.set(&iterator.storage_map_iterator.previous_key);
-		}
```

Comment:
I believe that the substrate iterator next() method will only return `None` in the case of the iterator finishing, hence the for loop instead of while let Some. 

I assume this was put like this for some reason but I can't tell why a for loop wasn't used in the first place